src/main/java/uk/ac/ic/wlgitbridge/bridge/context/ContextStore.java

```diff
@@ -98,12 +99,12 @@ public static ContextStore getInstance()  {
     ProjectContext context = instance.getContextForProject(projectName);
     try (ProjectLock lock = context.getLock()) {
       lock.lock();
-      Pair<R, Exception> result = f.apply(context);
+      R result = f.apply(context);
       lock.success();
       return result;
     } catch (Exception e) {
-      Log.error("[{}] Exception in context with lock", projectName, e);
-      return new Pair(null, e);
+      Log.error("[{}] Exception in context with lock: {}", projectName, e.getMessage());
+      throw e;
```

Comment:
This is a good improvement! I was suddenly worried that this might not preserve the stack trace from the original exception, but it does so that's fine. :+1: